frontend/src/app.js

```diff
@@ -186,7 +195,7 @@ search.addWidgets([
       { label: '45+ minutes', start: 2700 },
     ],
   }),
-  infiniteHits({
+  myWidgets.appInfiniteHits({
```

Comment:
I see and appreciate your tendency to refactor the code, however let's do refactoring as a separate task a bit later. Especially because this widget (infinite hits) is not related to the current task at all, therefore it is expected to be untouched in this PR.

Reply:
well my goal was not to refactor the code but fulfill the all three requirements you have asked. and its my natural style to not write clumsy code. i always try to write modular code.
second, if i had multiple options then i can think about which one is best. but as i told that i have zero knowledge and experience in current subject. so whatever i found useful i added in order to complete the task.

Reply:
That's good that you avoid clumsy code, I fully support this position. In this case I just don't see the necessity in overriding this widget.

Please accept this feedback of the developer who has some knowledge of Algolia. I realize that for you it's a new library, but together we can figure out the best option to implement the task.

Reply:
i respect you feedbacks, i am sorry if you are not satisfied with my answers.